src/main/kotlin/com/hitc/hitweconomyplugin/main/core/PlayerData.kt

```diff
@@ -1,9 +1,54 @@
 package com.hitc.hitweconomyplugin.main.core
 
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import java.io.File
+
 
 @Serializable
-data class PlayerData(
-    var credits : Int,
-    var creditsEarned : Int
-)
+class PlayerData(
+    private var credits: Int,
+    private var creditsEarned: Int,
+    @Transient var file : File? = null
+) {
+
+    fun addCredits(credits : Int) {
+        this.credits += credits
+        save()
+    }
+
+    fun subCredits(credits : Int) {
+        this.credits -= credits
+        save()
+    }
+
+    fun getCredits() : Int {
```

Comment:
it's quite dumb that we cannot use the base getter but i looked online and seems like it's either we do that and we can serialize or we can make the setters private but rip serialization.

I guess this is good as is